Kommunicate/Classes/KMConversationListViewController.swift

```diff
@@ -0,0 +1,642 @@
+            return
+        }
+
+        viewModel.updateDeliveryReport(convVC: viewController, messageKey: messageKey, contactId: contactId, status: status)
+    }
+
+    open func updateStatus(forContact contactId: String!, withStatus status: Int32) {
+        guard let viewController = conversationViewController ?? conversationVC(), viewController.viewModel != nil else {
+            return
+        }
+
+        viewModel.updateStatusReport(convVC: viewController, forContact: contactId, status: status)
+    }
+
+    open func updateTypingStatus(_: String!, userId: String!, status: Bool) {
+        print("Typing status is", status)
+
+        guard let viewController = conversationViewController ?? conversationVC(), let vm = viewController.viewModel else { return
+        }
+        guard (vm.contactId != nil && vm.contactId == userId) || vm.channelKey != nil else {
+            return
+        }
+        print("Contact id matched")
+        viewModel.updateTypingStatus(in: viewController, userId: userId, status: status)
+    }
+
+    open func reloadData(forUserBlockNotification userId: String!, andBlockFlag _: Bool) {
+        print("reload data")
+        let userDetail = ALUserDetail()
+        userDetail.userId = userId
+        viewModel.updateStatusFor(userDetail: userDetail)
+        guard let viewController = navigationController?.visibleViewController as? KMConversationViewController else {
+            return
+        }
+        viewController.checkUserBlock()
+    }
+
+    open func updateLastSeen(atStatus alUserDetail: ALUserDetail!) {
+        print("Last seen updated")
+        viewModel.updateStatusFor(userDetail: alUserDetail)
+        guard let viewController = navigationController?.visibleViewController as? KMConversationViewController else {
+            return
+        }
+        viewController.updateLastSeen(atStatus: alUserDetail)
+    }
+
+    open func mqttConnectionClosed() {
+        print("ALKConversationListVC mqtt connection closed.")
+        alMqttConversationService.retryConnection()
+    }
+}
+
+extension KMConversationListViewController: ALKConversationListTableViewDelegate {
+    public func tapped(_ chat: ALKChatViewModelProtocol, at index: Int) {
+
+        let convViewModel = conversationViewModelType.init(contactId: chat.contactId, channelKey: chat.channelKey, localizedStringFileName: configuration.localizedStringFileName)
+        let convService = ALConversationService()
+        if let convId = chat.conversationId, let convProxy = convService.getConversationByKey(convId) {
+            convViewModel.conversationProxy = convProxy
+        }
+        let viewController = conversationViewController ?? KMConversationViewController(configuration: configuration)
+        viewController.viewModel = convViewModel
+        navigationController?.pushViewController(viewController, animated: false)
```

Comment:
We can remove `conversationId` part as we are not using that. And use existing `launchChat` method for launch.